### PR TITLE
Use Java version of Google Closure Compiler

### PR DESCRIPTION
**Pull request by gaearon**
This was actually done by @trueadm, I just did a minor cleanup on top while he's away from keyboard.

The JS version of GCC is already too slow, and **its memory usage is getting worse as we throw more inlinable code at it**. Without this change, we can't land https://github.com/facebook/react/pull/12792 (7% prod build size reduction) because it eats over 16GB of RAM and never completes. For comparison, the Java version memory use peaks under 300 MB and completes in a few seconds. Luckily it's nicely packaged under a launcher distributed on npm so it's not a pain to integrate.

This change, however, means you can't `yarn build` without Java installed. I can imagine this being annoying for open source contributors. I don't see better options though.

### CODE REVIEW

**Comment by react-linter** [comment]
<details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 37d12e29169f4ef4ab06eb0626cde4f0fee729cc...8478f2dcd8feff2e768f16d2f131216c5bfbf16e</p>


  
## react-noop-renderer
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-noop-renderer.development.js | +2.8% | +8.8% | 17.86 KB | 18.35 KB | 4.69 KB | 5.1 KB | NODE_DEV
**react-noop-renderer.production.min.js** | **-4.1%** | **-2.3%** | **6.62 KB** | **6.35 KB** | **2.56 KB** | **2.5 KB** | **NODE_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: scripts/rollup/build.js [hunk 1]
```diff
@@ -473,7 +473,7 @@ function handleRollupError(error) {
       highlightCode: true,
     });
     console.error(frame);
-  } else {
+  } else if (error.codeFrame) {
```
Comment (bvaughn) [inline scripts/rollup/build.js]: Should we add an `else` here with something (e.g. "Unknown error")?

Comment (gaearon) [reply scripts/rollup/build.js]: We already print the error message itself earlier so these are additive.

**Comment by gaearon** [comment]
CircleCI part works (I tested), as for AppVeyor we'll have to see because it only runs on master.

**Comment by sophiebits** [comment]
Should we file a bug with Closure?

**Comment by gaearon** [comment]
Nah, they're aware JS version is vastly inferior and were actually surprised it was usable to us. 😄

**Pull request merged**